src/client/js/Widgets/DiagramDesigner/DiagramDesignerWidget.js

```diff
@@ -206,16 +207,10 @@ define([
 
         /*********** CONNECTION DRAWING COMPONENT *************/
         this._defaultConnectionRouteManagerType = params.defaultConnectionRouteManagerType;
-        //initiate Connection Router (if needed)
-        if (params.connectionRouteManager) {
-            this.connectionRouteManager = params.connectionRouteManager;
-        } else if (params.defaultConnectionRouteManagerType === 'basic') {
-            this.connectionRouteManager = new ConnectionRouteManagerBasic({diagramDesigner: this});
-        } else if (params.defaultConnectionRouteManagerType === 'basic2') {
-            this.connectionRouteManager = new ConnectionRouteManager2({diagramDesigner: this});
-        } else if (params.defaultConnectionRouteManagerType === 'basic3') {
-            this.connectionRouteManager = new ConnectionRouteManager3({diagramDesigner: this});
-        }
+
+        //initiate Connection Router
+        router = params.connectionRouteManager || params.defaultConnectionRouteManagerType;
+        this._onConnectionRouteManagerChanged(router);
```

Comment:
Wouldn't this change the behavior if params.connectionRouteManager is passed?
That's an instance of a RouteManager and in _onConnectionRouteManagerChanged the switch statement would always reset it to ConnectionRouteManagerBasic.